mllib/src/test/scala/org/apache/spark/mllib/clustering/GaussianMixtureSuite.scala

```diff
@@ -148,6 +148,30 @@ class GaussianMixtureSuite extends SparkFunSuite with MLlibTestSparkContext {
     }
   }
 
+  test("model prediction, parallel and local") {
+
+    val data = sc.parallelize(GaussianTestData.data)
+
+    val gmm = new GaussianMixture().setK(2).setSeed(0).run(data)
+
+    val batchPredictions = gmm.predict(data).collect()
```

Comment:
Be careful about assuming rows are returned in the same order.  It generally works but is not guaranteed.  Instead, how about doing:
```
val batchPredictions = gmm.predict(data)
batchPredictions.zip(data).collect().foreach { case (batchPred, datum) =>
  assert(batchPred === gmm.predict(datum))
}
```
Note the use of triple equals (better type handling).

Same for soft predictions.